File: annotations/src/main/java/com/siimkinks/sqlitemagic/annotation/IgnoreColumn.java
package com.siimkinks.sqlitemagic.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * All fields with this annotation are ignored when {@link Table#persistAll()} is true on
 * {@link Table} annotation.<br>
 * It has no other functionality.
 */
@Target({ElementType.FIELD, ElementType.METHOD})
@Retention(RetentionPolicy.CLASS)
public @interface IgnoreColumn {
}

File: annotations/src/main/kotlin/com/siimkinks/sqlitemagic/annotation/IgnoreColumn.kt
package com.siimkinks.sqlitemagic.annotation

import kotlin.annotation.AnnotationRetention.BINARY
import kotlin.annotation.AnnotationTarget.FIELD

/**
 * Excludes a property from persistence when [Table.persistAll] is `true`.
 *
 * This annotation has no other functionality.
 */
@Target(FIELD)
@Retention(BINARY)
annotation class IgnoreColumn
